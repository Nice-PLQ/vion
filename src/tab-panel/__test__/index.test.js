import { mount } from '@vue/test-utils';
import VTabPanel from '../tab-panel.vue';
import VTab from '../../tab';

describe('TabPanel', () => {
  test('渲染', () => {
    const wrapper = mount({
      template: `
        <v-tab>
          <v-tab-panel v-for="(tab,index) in tabs" :title="tab" :key="index">{{tab}}</v-tab-panel>
        </v-tab>
      `,
      data() {
        return {
          tabs: [
            '热门赛事',
            'QQ飞车手游',
            '英雄联盟',
            '王者荣耀',
            '和平精英'
          ]
        };
      }
    }, {
      global: {
        components: { VTab, VTabPanel }
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
