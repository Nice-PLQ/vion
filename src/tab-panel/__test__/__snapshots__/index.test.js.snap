// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel 渲染 1`] = `
<div
  class="v-tab"
>
  <div
    class="v-tab__nav-wrap"
  >
    <ul
      class="v-tab__nav"
      role="tablist"
      style="transform: translate3d(0px, 0, 0);"
    >
      
      
    </ul>
  </div>
  <div
    class="v-tab__content"
    style="transform: translate3d(0px, 0, 0); transition: all 400ms ease;"
  >
    
    
    <div
      class="v-tab__panel"
    >
      
      热门赛事
      
    </div>
    <div
      class="v-tab__panel"
    >
      
      QQ飞车手游
      
    </div>
    <div
      class="v-tab__panel"
    >
      
      英雄联盟
      
    </div>
    <div
      class="v-tab__panel"
    >
      
      王者荣耀
      
    </div>
    <div
      class="v-tab__panel"
    >
      
      和平精英
      
    </div>
    
    
  </div>
</div>
`;
